api/apps/api/src/modules/scenarios/scenarios.controller.ts

```diff
@@ -1215,4 +1220,59 @@ export class ScenariosController {
 
     return result.right;
   }
+
+  @ApiOperation({ description: 'Get PDF summary report for scenario' })
+  @ApiOkResponse()
+  @Header('content-type', 'application/pdf')
+  @Post('/:scenarioId/solutions/report')
+  async getSummaryReportForProject(
+    @Body() config: WebshotConfig,
+    @Param('scenarioId', ParseUUIDPipe) scenarioId: string,
+    @Res() res: Response,
+    @Req() req: RequestWithAuthenticatedUser,
+    @AppSessionToken() appSessionToken: string,
+  ): Promise<any> {
+    /**
+     * If a frontend app session token was provided via cookie, use this to let
+     * the webshot service authenticate to the app, otherwise fall back to
+     * looking for the relevant cookies in the body of the request.
+     *
+     * @todo Remove this once the new auth workflow via `Cookie` header is
+     * stable.
+     */
+    const configForWebshot = appSessionToken
+      ? {
+          ...config,
+          cookie: appSessionToken,
+        }
+      : config;
+    // @debt Refactor to use @nestjs/common's StreamableFile
+    // (https://docs.nestjs.com/techniques/streaming-files#streamable-file-class)
+    // after upgrading NestJS to v8.
+    const scenario = await this.service.getById(scenarioId, {
+      authenticatedUser: req.user,
+    });
+
+    if (isLeft(scenario)) {
+      throw mapAclDomainToHttpError(scenario.left, {
+        scenarioId,
+        userId: req.user.id,
+        resourceType: scenarioResource.name.plural,
+      });
+    }
+
+    const pdfStream = await this.webshotService.getSummaryReportForScenario(
+      scenarioId,
+      scenario.right.projectId,
+      configForWebshot,
+    );
```

Comment:
what do you think about moving this into a small function in `ScenariosService`? this way we avoid adding ACL logic in the controller (though I think there will be a hopefully manageable circular dependency lurking if we do so)

Reply:
I mean, at least the `getById()` and `if(isLeft(scenario))` bits - the pdfstream thingy can stay, but basically it would use `this.scenariosService` which would pass arguments through to `webshotService`, and in turn pass back the stream from `webshotService`.

Reply:
ok, I see you've dealt with circular deps between scenarios service and webshot service already below

Reply:
as discussed, I've done this - 5175ac5aa66ad5923dea5aafba7322fe5a72c4af